src/block_disagg/block_disagg_read.c

```diff
@@ -52,6 +52,27 @@ __block_disagg_read_checksum_err(WT_SESSION_IMPL *session, const char *name, uin
       name, size, page_id, lsn, context_msg, checksum, expected_checksum);
 }
 
+/*
+ * __block_disagg_check_lsn_frontier --
+ *     Check that the LSN is not ahead of the materialization frontier.
+ */
+static void
+__block_disagg_check_lsn_frontier(WT_SESSION_IMPL *session, uint64_t lsn)
+{
+    uint64_t last_materialized_lsn;
+
+    WT_ACQUIRE_READ(
+      last_materialized_lsn, S2C(session)->disaggregated_storage.last_materialized_lsn);
+    if (last_materialized_lsn != WT_DISAGG_LSN_NONE && lsn > last_materialized_lsn) {
+        /* FIXME-WT-15818 Consider crashing upon this check failure. */
+        WT_STAT_CONN_INCR(session, disagg_block_read_ahead_frontier);
+        __wt_verbose_error(session, WT_VERB_DISAGGREGATED_STORAGE,
```

Comment:
Chatted with Kaitlin Mahar and she suggested that Server's tests shouldn't fail on unexpected output like this, so keeping it an error for visibility.
Feel free to reconsider the message's level later.